sql/plan/alter_table.go

```diff
@@ -145,7 +146,105 @@ func (a *AddColumn) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error)
 		return nil, err
 	}
 
-	return sql.RowsToRowIter(), alterable.AddColumn(ctx, a.column, a.order)
+	err = alterable.AddColumn(ctx, a.column, a.order)
+	if err != nil {
+		return nil, err
+	}
+
+	return sql.RowsToRowIter(), a.updateRowsWithDefaults(ctx, row)
+}
+
+// updateRowsWithDefaults iterates through an updatable table and applies an update to each row.
+func (a *AddColumn) updateRowsWithDefaults(ctx *sql.Context, row sql.Row) error {
+	updatable, err := getUpdatable(a.Child)
+	if err != nil {
+		return err
+	}
+
+	tableIter, err := a.Child.RowIter(ctx, row)
+	if err != nil {
+		return err
+	}
+
+	schema := updatable.Schema()
+	idx := -1
+	for i, col := range schema {
+		if col.Name == a.column.Name {
+			idx = i
+		}
+	}
+
+	updater := updatable.Updater(ctx)
+
+	for {
+		r, err := tableIter.Next(ctx)
+		if err == io.EOF {
+			return updater.Close(ctx)
+		}
+
+		if err != nil {
+			_ = updater.Close(ctx)
+			return err
+		}
+
+		updatedRow, err := applyDefaults(ctx, schema, idx, r, a.column.Default)
+		if err != nil {
+			return err
+		}
+
+		err = updater.Update(ctx, r, updatedRow)
+		if err != nil {
+			return err
+		}
+	}
+}
+
+// applyDefaults applies the default values of the given column indices to the given row, and returns a new row with the updated values.
+// This assumes that the given row has placeholder `nil` values for the default entries, and also that each column in a table is
+// present and in the order as represented by the schema.
+func applyDefaults(ctx *sql.Context, tblSch sql.Schema, col int, row sql.Row, cd *sql.ColumnDefaultValue) (sql.Row, error) {
+	newRow := row.Copy()
+	if len(tblSch) != len(row) {
+		return nil, fmt.Errorf("any row given to ApplyDefaults must be of the same length as the table it represents")
+	}
+	secondPass := false
+	if col < 0 || col > len(tblSch) {
+		return nil, fmt.Errorf("column index `%d` is out of bounds, table schema has `%d` number of columns", col, len(tblSch))
+	}
+
+	columnDefaultExpr := cd
+	if !columnDefaultExpr.IsLiteral() {
+		secondPass = true
+	} else if columnDefaultExpr == nil && !tblSch[col].Nullable {
+		val := tblSch[col].Type.Zero()
+		var err error
+		newRow[col], err = tblSch[col].Type.Convert(val)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		val, err := columnDefaultExpr.Eval(ctx, newRow)
+		if err != nil {
+			return nil, err
+		}
+		newRow[col], err = tblSch[col].Type.Convert(val)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if secondPass {
```

Comment:
secondPass is actually a no-op in this version of the code (Daylon pointed this out to me on my own PR), get rid of it